lib/Config.js

```diff
@@ -192,32 +192,34 @@ function hdClientLocationConstraintAssert(configHd) {
 }
 
 function dmfLocationConstraintAssert(locationObj) {
-    const {
-        endpoint,
-        username,
-        password,
-        repoId,
-        nsId,
-    } = locationObj.details;
-    const stringFields = [
-        endpoint,
-        username,
-        password,
-        nsId,
-    ];
-    stringFields.forEach(field => {
-        assert(typeof field === 'string',
-            `bad config: ${field} must be a string`);
-    });
-    assert.strictEqual(Array.isArray(repoId));
-    repoId.forEach(rId => {
-        assert(typeof rId === 'string',
-            `bad config: ${rId} must be a string`);
-    });
-    assert(repoId.every(
-        r => typeof r === 'string'),
-        'bad config: each repoId must be a string',
-    );
+    if (Object.keys(locationObj.details).length > 0) {
```

Comment:
I would switch to a guard clause, no need for the extra tabbing and easier to remove later
```suggestion
    if (Object.keys(locationObj.details).length === 0) {
        return
    }
```

Reply:
Agreed, thanks as always :smile:

Reply:
Done: https://github.com/scality/cloudserver/pull/4543/commits/44ad45ab36913a06a60b4b7015fd6a0d5faf5b14